bin/akula-rpc.rs

```diff
@@ -43,15 +124,572 @@ where
+            block_body::read_with_senders(&self.db.begin().await?, block_hash, block_number)
+                .await?
+                .unwrap();
+
+        let mut analysis_cache = AnalysisCache::default();
+        let mut engine = engine_factory(MAINNET.clone()).unwrap();
+        let block_spec = MAINNET.collect_block_spec(block_number);
+        let mut processor = ExecutionProcessor::new(
+            &mut state,
+            None,
+            &mut analysis_cache,
+            &mut *engine,
+            partial_header,
+            &block,
+            &block_spec,
+        );
+
+        let receipt = processor.execute_block_no_post_validation().await?[index].clone();
+
+        let mut logs: Vec<TxLog> = Vec::new();
+        let mut i = 0;
+        while i < receipt.logs.len() {
+            logs.push(TxLog {
+                log_index: Some(U64::from(i)),
+                transaction_index: Some(U64::from(index)),
+                transaction_hash: Some(tx_hash),
+                block_hash: Some(block_hash),
+                block_number: Some(U64::from(block_number.0)),
+                address: receipt.logs[i].address,
+                data: receipt.logs[i].data.clone(),
+                topics: receipt.logs[i].topics.clone(),
+            });
+            i += 1;
+        }
+
+        let to = match msg.action() {
+            TransactionAction::Call(to) => Some(to),
+            TransactionAction::Create => None,
+        };
+
+        let contract_address = match to {
+            Some(_) => None,
+            None => Some(create_address(msg.sender, msg.nonce())),
+        };
+
+        Ok(TxReceipt {
+            transaction_hash: tx_hash,
+            transaction_index: U64::from(index),
+            block_hash,
+            block_number: U64::from(block_number.0),
+            from: msg.sender,
+            to,
+            cumulative_gas_used: U64::from(receipt.cumulative_gas_used),
+            gas_used: U64::from(header.gas_used),
+            contract_address,
+            logs,
+            logs_bloom: receipt.bloom,
+            status: U64::from(receipt.success as u64),
+        })
+    }
+
+    async fn get_uncle_by_block_hash_and_index(
+        &self,
+        block_hash: H256,
+        index: UncleIndex,
+    ) -> RpcResult<jsonrpc::common::Block> {
+        let block_number = header_number::read(&self.db.begin().await?, block_hash)
+            .await?
+            .unwrap();
+
+        let block_body =
+            block_body::read_without_senders(&self.db.begin().await?, block_hash, block_number)
+                .await?
+                .unwrap();
+
+        let ommer_hash = block_body.ommers[index.as_u64() as usize].hash();
+
+        let ommer_block_number = header_number::read(&self.db.begin().await?, ommer_hash)
+            .await?
+            .unwrap();
+
+        Ok(json_obj::assemble_block(
+            &self.db.begin().await?,
+            ommer_hash,
+            ommer_block_number,
+            true,
```

Comment:
[The docs](https://eth.wiki/json-rpc/API#eth_getunclebyblockhashandindex) don't specify if a list of full tx objects or just a list of tx hashes should be returned. Which should we use?

Reply:
If I read the docs correctly, it says that the return is the same as eth_getBlockByHash, i.e. block objects. If in doubt, you can run a geth or erigon console, and check what's returned.

Reply:
Ok, I will return the same format as erigon and geth. I think you have misunderstood the question, it refers to the transactions within a block object:
**transactions: Array** - Array of transaction objects, or 32 Bytes transaction hashes depending on the last given parameter.